client/components/theme-tier/theme-tier-badge/theme-tier-upgrade-badge.js

```diff
@@ -2,25 +2,54 @@ import { getPlan } from '@automattic/calypso-products';
 import { PremiumBadge } from '@automattic/components';
 import { Plans } from '@automattic/data-stores';
 import { createInterpolateElement } from '@wordpress/element';
+import clsx from 'clsx';
 import { useTranslate } from 'i18n-calypso';
+import { useGoalsFirstExperiment } from 'calypso/landing/stepper/declarative-flow/helpers/use-goals-first-experiment';
+import { useSelector } from 'calypso/state';
 import { useThemeTierForTheme } from 'calypso/state/themes/hooks/use-theme-tier-for-theme';
+import { getMarketplaceThemeSubscriptionPrices } from 'calypso/state/themes/selectors';
 import { THEME_TIERS } from '../constants';
 import ThemeTierBadgeCheckoutLink from './theme-tier-badge-checkout-link';
 import { useThemeTierBadgeContext } from './theme-tier-badge-context';
 import ThemeTierTooltipTracker from './theme-tier-tooltip-tracker';
 
-export default function ThemeTierUpgradeBadge() {
+export default function ThemeTierPlanUpgradeBadge( { showPartnerPrice } ) {
 	const translate = useTranslate();
 	const { themeId } = useThemeTierBadgeContext();
 	const themeTier = useThemeTierForTheme( themeId );
 
 	const tierMinimumUpsellPlan = THEME_TIERS[ themeTier?.slug ]?.minimumUpsellPlan;
 	const mappedPlan = getPlan( tierMinimumUpsellPlan );
 	const planPathSlug = mappedPlan?.getPathSlug();
+	const subscriptionPrices = useSelector( ( state ) =>
+		getMarketplaceThemeSubscriptionPrices( state, themeId )
+	);
 
 	// Using API plans because the updated getTitle() method doesn't take the experiment assignment into account.
 	const plans = Plans.usePlans( { coupon: undefined } );
 	const planName = plans?.data?.[ mappedPlan.getStoreSlug() ]?.productNameShort;
+	const [ , isGoalsAtFrontExperiment ] = useGoalsFirstExperiment();
+
+	const getLabel = () => {
+		if ( ! isGoalsAtFrontExperiment ) {
+			return translate( 'Upgrade' );
+		}
+
+		if ( showPartnerPrice && subscriptionPrices.month ) {
+			return translate( 'Available on %(planName)s plus %(price)s/month', {
+				args: {
+					planName: planName,
+					price: subscriptionPrices.month,
+				},
+			} );
+		}
```

Comment:
This is the logic for rendering the subscription price on the design page @fditrapani.

We're rendering the monthly price, which is usually higher than the annual price divided by 12.

I believe it'd be better to show the cheapest version, but it would become quite long if we said `Available on Business plus $7/month or $59/year`.

Reply:
"Available on Business plus $5/month (billed annually)" ??

I think we should merge with this copy we have though. Showing the price "yearly divided by 12" is probably better, if we can come up with some good copy. But that's the sort of optimisation we can make in a follow up IMO